pkg/util/logs/logs.go

```
@@ -4,41 +4,28 @@
 package logs
 
 import (
-	"flag"
-	"log"
-	"time"
-
-	"github.com/golang/glog"
-	"github.com/spf13/pflag"
-	"k8s.io/apimachinery/pkg/util/wait"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-var logFlushFreq = pflag.Duration("log-flush-frequency", 5*time.Second, "Maximum number of seconds between log flushes")
-
-// GlogWriter serves as a bridge between the standard log package and the glog package.
-type GlogWriter struct{}
-
-// Write implements the io.Writer interface.
-func (writer GlogWriter) Write(data []byte) (n int, err error) {
-	glog.Info(string(data))
-	return len(data), nil
-}
-
-// InitLogs initializes logs the way we want for kubernetes.
-func InitLogs() {
-	log.SetOutput(GlogWriter{})
-	log.SetFlags(0)
-	flag.Set("logtostderr", "true")
-	// The default glog flush interval is 30 seconds, which is frighteningly long.
-	go wait.Until(glog.Flush, *logFlushFreq, wait.NeverStop)
-}
-
-// FlushLogs flushes logs immediately.
-func FlushLogs() {
-	glog.Flush()
+//VerbosityToSeverity converts verbosity-level to zap log-level
+func VerbosityToSeverity(verbosity int) zapcore.Level {
+	if verbosity > 4 { //Most verbose
+		return zapcore.DebugLevel
+	}
+	if verbosity < 0 { //Least verbose
+		return zapcore.FatalLevel
+	}
+	return zapcore.Level(4 - verbosity)
 }
 
-// NewLogger creates a new log.Logger which sends logs to glog.Info.
-func NewLogger(prefix string) *log.Logger {
-	return log.New(GlogWriter{}, prefix, 0)
+// InitLogs initializes logs with Time and Global Level of Logs set at Info
+func InitLogs(verbosity int) {
+	config := zap.NewProductionConfig()
+	config.Level.SetLevel(VerbosityToSeverity(verbosity))
+	config.EncoderConfig.EncodeTime = zapcore.RFC3339TimeEncoder
+	config.EncoderConfig.TimeKey = "time"
+	config.EncoderConfig.MessageKey = "message"
+	logger, _ := config.Build()
```

This is ignoring any errors - they need to be handled.

---

fixed